tinygrad/shape/view.py

```diff
@@ -127,6 +174,20 @@ def reshape(self, new_shape: Tuple[sint, ...]) -> Optional[View]:
           new_mask_tuple = tuple([(0,1) if x == 1 else new_mask.pop(0) for x in new_shape])
       return View.create(new_shape, new_strides_tuple, self.offset, new_mask_tuple)
 
-    # TODO: bring the merge_views logic here for more caching
+    # TODO: this is incomplete with symbolic expand (3,i) -> (3,i,1) so still need the adding/removing 1s part
```

Comment:
can reduce some lines to fix this

Reply:
```
i = Variable("i", 1, 3).bind(3)
x = View.create((4,i), mask = ((0,2),(0,i)))
y = x.reshape((4,i,1))
```

this seems to be working.

I removed the add/remove ones logic. Even then while running winograd tests, the call to merge_views is 2 only.

Reply:
Seems like the add/remove ones cases are being handled by the reshape_mask logic. Any case you have where it doesn't work ?

Reply:
@chenyuxyz have a look. did this by slight modification in the accumulate while loop.

```
view = View.create(shape=(2,4,5,5),strides=(4,5,1,3))
st = ShapeTracker(views=(view,))
st.views[-1].reshape((1,2,1,4,5,1,5,1,1)) # works without returning none
```